base/doc/usrguide.tex

```diff
@@ -1258,4 +1258,39 @@ \section{Case changing}
   |\DeclareUppercaseMapping[xx]{"0131}{I}|
 \end{quote}
 
+\section{Support for problem solving}
+
+\begin{decl}
+  |\listfiles| \oarg{options}
+\end{decl}
+
+If this command is placed in the preamble then a list of the files
+read in (as a result of processing the document) will be displayed
+on the terminal (and in the log file) at the end of the run. Where
+possible, a short description will also be produced. These descriptions
+will (hopefully) include the descriptions, dates and version numbers
+for package and class files.
+
+Sometimes, if may be that a local edit has been made to a package or
+class file (or rather a copy of such a file). To allow these cases to
+be identified, \cs{listfiles} takes an optional argument which allows
+adjustment of the information printed using a key--value approach
+\begin{description}
+  \item[\texttt{hashes}] Adds the MD5 hash for each file to the
+    information printed
+  \item[\texttt{sizes}] Adds the file size for each file to the
+    information printed
+\end{description}
+Note that the difference in line ends used by Windows and Unix
+systems \emph{does} show up here: as such, you should compare
+results from computers using the same operating system.
+
+\emph{Warning}: this command will list only files which were read
+using \LaTeX{} commands such as |\input|\arg{file} or
+|\include|\arg{file}.  If the file was read using the primitive \TeX{}
+syntax |\input |\emph{file} (without |{ }| braces around the file name)
+then it will not be listed; failure to use the \LaTeX{} form with the
```

Comment:
Starting from TeX Live 2020, `\input` primitive "also accepts a group-delimited filename argument".
https://tug.org/texlive/doc/texlive-en/texlive-en.html#x1-890009.1.17

Reply:
Sure, but in LaTeX a braced argument will always go _via_ the LaTeX mech: the point is that anything which doesn't use braces is a problem.

Reply:
I think it is fine as it is given what you want to convey